### PR TITLE
implement is_locked function

### PR DESCRIPTION
**Pull request by casweeney**
## `is_locked()` function implementation - closes: https://github.com/Token-Giver/token-giver-contracts/issues/47
- updated `Scarb.toml` file, installed Openzeppelin `v0.17.0` and token_bound_accounts `v0.3.0`.
- Implemented `is_locked()` function using the `ILockableDispatcher` and `campaign_address`.
- Wrote test `test_is_locked()` for the implemented `is_locked()` function.

### CODE REVIEW

File: Scarb.lock [hunk 1]
```diff
@@ -3,13 +3,16 @@ version = 1
```
Comment (mubarak23) [inline Scarb.lock:3]: please remove the scarb.lock file

File: src/campaign.cairo [hunk 2]
```diff
@@ -27,6 +27,9 @@ mod TokengiverCampaign {
     use openzeppelin::access::ownable::OwnableComponent;
     use openzeppelin::upgrades::UpgradeableComponent;
     use openzeppelin::upgrades::interface::IUpgradeable;
+    use token_bound_accounts::interfaces::ILockable::{
```
Comment (mubarak23) [inline src/campaign.cairo:30]: So you only import the component you are interested in,

nice 
Good Work

**Review by mubarak23 — CHANGES_REQUESTED** [review]
This is Good

Well Done, 

remove the scarb.lock and i will merged it.

Thank you.

File: src/campaign.cairo (hunk 2 above)
Comment (casweeney) [reply src/campaign.cairo:30]: Thanks big chief.

File: Scarb.lock (hunk 1 above)
Comment (casweeney) [reply Scarb.lock]: > please remove the scarb.lock file

Done 💯

**Comment by casweeney** [comment]
> This is Good
> 
> Well Done,
> 
> remove the scarb.lock and i will merged it.
> 
> Thank you.

@mubarak23 thanks for your feedback. `Scarb.lock` has been removed. I noticed the test fails here, I looked at it, it's an RPC error from the CI, not from the code. 

Just thought I mention it 🫡 .

**Pull request merged**